packages/linter/src/generators/lint-project/lint-project.ts

```diff
@@ -51,7 +51,7 @@ export async function lintProjectGenerator(
   const projectConfig = readProjectConfiguration(tree, options.project);
 
   const lintFilePatterns = options.eslintFilePatterns;
```

Comment:
This would be the accepted solution since the option **is marked** as optional

```suggestion
  const lintFilePatterns = options.eslintFilePatterns ?? [];
```